### ISSUE TITLE
[feature request] Hooks after accumulating all the gradients

### ISSUE THREAD

**Issue by Stonesjtu** [pytorch/pytorch#8023]
Currently we can `register_hook` so a customized function is going to execute every time a gradient is computed. But what if I want to run a hook function only when the total **gradient is accumulated**? 
Such hook should be helpful when I want to overlap the transfer latency with backward on distributed training.
I can't find any documentation for that. And I think the *hook system* can potentially be much more powerful.

**Comment by albanD** [comment]
Hi,

You can just register your hook on the tensor where the gradients are accumulated. The hook will be called after all gradients for this tensors have been computed.

**Comment by Stonesjtu** [comment]
Thx for helping.
Is this feature documented, I can't find any resources for it.

**Comment by apaszke** [comment]
It's [right here](https://pytorch.org/docs/stable/autograd.html?highlight=register_hook#torch.Tensor.register_hook).
